### DISCUSSION TITLE
Merging/concatenating nested JSON objects

### DISCUSSION THREAD

**Discussion by linusmossberg** [nlohmann/json#2700] [Q&A] [answered]
Hi!

I'm looking for a generalized way of merging two JSON objects, as shown in the following example:

`j1`:
```json
{
  "A": [ {  "x": 1  },  { "x": 2 } ],

  "B":{
    "C": [ 1, 2 ],
    "D": 5,
    "E": { 
      "F": [ 1, 2 ]
    }
  }
}
```

`j2`:
```json
{
  "A": [ { "x": 3 } ],

  "B": {
    "C": [ 3, 4 ],
    "D": 6,
    "E": { 
      "F": [ 3, 4 ],
      "G": "ABC"
    },
    "H": 6
  }
}
```

Expected result of merging `j1` and `j2`:
```json
{
  "A": [ {  "x": 1  },  { "x": 2 }, { "x": 3 } ],

  "B": {
    "C": [ 1, 2, 3, 4 ],
    "D": 6,
    "E": {
      "F": [ 1, 2, 3, 4 ],
      "G": "ABC"
    },
    "H": 6
  }
}
```

Is this possible? I've tried using `update` and `insert`, but this seems to overwrite key values rather than concatenating them when the value is e.g. an array. The object that takes precedence when it makes sense to overwrite values is not important.

**Comment by nlohmann** [comment, marked as answer]
No, there exists no such function in the library, but I am confident that you can implement it with the public API.

**Reply by linusmossberg** [reply]
Ok, thanks for the quick reply! Yes, I guess iterating through each key, keeping track of the level and setting up a switch case for key value types may work.
